src/main/scala/wdl4s/MemorySize.scala

```diff
@@ -0,0 +1,39 @@
+package wdl4s
+
+import wdl4s.parser.MemoryUnit
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+object MemorySize {
+  val memoryPattern = """(\d+(?:\.\d+)?)\s*(\w+)""".r
+  def parse(unparsed: String): Try[MemorySize] = {
```

Comment:
@cjllanwarne I don't care about this enough to continue the argument but my pushback on your point is this -> I think on the surface you're exactly right, however that Try in the signature provides a false sense of security as there's 0 guarantee it's not actually throwing some other exception on you. So then what do you do, wrap that response in its own Try?